modules/core/src/utils/prop-equal.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Fast partial deep equal for prop.
+ *
+ * @param a Prop
+ * @param b Prop to compare against `a`
+ * @param depth Depth to which to recurse in nested Objects/Arrays. Use 0 (default) for shallow comparison, -1 for infinite depth
+ */
+/* eslint-disable complexity */
+export function propEqual(a: any, b: any, depth: number): boolean {
```

Comment:
I personally think it's fine to call this function `deepEqual`.

Reply:
OK, reverted back to old naming